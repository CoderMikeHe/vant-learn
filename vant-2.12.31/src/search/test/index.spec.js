import Search from '..';
import { mount } from '../../../test';

test('input event', () => {
  const onInput = jest.fn();
  const wrapper = mount(Search, {
    context: {
      on: {
        input: onInput,
      },
    },
  });

  const input = wrapper.find('input');
  input.element.value = '1';
  input.trigger('input');

  expect(onInput).toHaveBeenCalledWith('1');
});

test('cancel event', () => {
  const onInput = jest.fn();
  const onCancel = jest.fn();

  const wrapper = mount(Search, {
    propsData: {
      value: 'test',
      showAction: true,
    },
    context: {
      on: {
        input: onInput,
        cancel: onCancel,
      },
    },
  });

  const cancel = wrapper.find('.van-search__action');
  cancel.trigger('click');

  expect(onInput).toHaveBeenCalledWith('');
  expect(onCancel).toHaveBeenCalled();
});

test('search event', () => {
  const onSearch = jest.fn();
  const onKeypress = jest.fn();

  const wrapper = mount(Search, {
    context: {
      on: {
        search: onSearch,
        keypress: onKeypress,
      },
    },
  });

  const input = wrapper.find('input');
  input.trigger('keypress.enter');
  input.trigger('keypress.a');

  expect(onSearch).toHaveBeenCalled();
  expect(onKeypress).toHaveBeenCalled();
});

test('label slot', () => {
  const wrapper = mount(Search, {
    scopedSlots: {
      label: () => 'Custom Label',
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('left slot', () => {
  const wrapper = mount(Search, {
    scopedSlots: {
      left: () => 'Custom Left Content',
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('left-icon prop', () => {
  const wrapper = mount(Search, {
    propsData: {
      leftIcon: 'setting-o',
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('right-icon prop', () => {
  const wrapper = mount(Search, {
    propsData: {
      rightIcon: 'setting-o',
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('right-icon slot', () => {
  const wrapper = mount(Search, {
    scopedSlots: {
      'right-icon': () => 'Custom Right Icon',
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('action-text prop', () => {
  const wrapper = mount(Search, {
    propsData: {
      actionText: 'Custom Text',
      showAction: true,
    },
  });

  expect(wrapper).toMatchSnapshot();
});
