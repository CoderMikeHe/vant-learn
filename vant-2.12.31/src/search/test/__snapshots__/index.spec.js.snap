// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action-text prop 1`] = `
<div class="van-search van-search--show-action">
  <div class="van-search__content van-search__content--square">
    <div class="van-cell van-cell--borderless van-field">
      <div class="van-field__left-icon"><i class="van-icon van-icon-search">
          <!----></i></div>
      <div class="van-cell__value van-cell__value--alone van-field__value">
        <div class="van-field__body"><input type="search" class="van-field__control"></div>
      </div>
    </div>
  </div>
  <div role="button" tabindex="0" class="van-search__action">Custom Text</div>
</div>
`;

exports[`label slot 1`] = `
<div class="van-search">
  <div class="van-search__content van-search__content--square">
    <div class="van-search__label">Custom Label</div>
    <div class="van-cell van-cell--borderless van-field">
      <div class="van-field__left-icon"><i class="van-icon van-icon-search">
          <!----></i></div>
      <div class="van-cell__value van-cell__value--alone van-field__value">
        <div class="van-field__body"><input type="search" class="van-field__control"></div>
      </div>
    </div>
  </div>
</div>
`;

exports[`left slot 1`] = `
<div class="van-search">Custom Left Content<div class="van-search__content van-search__content--square">
    <div class="van-cell van-cell--borderless van-field">
      <div class="van-field__left-icon"><i class="van-icon van-icon-search">
          <!----></i></div>
      <div class="van-cell__value van-cell__value--alone van-field__value">
        <div class="van-field__body"><input type="search" class="van-field__control"></div>
      </div>
    </div>
  </div>
</div>
`;

exports[`left-icon prop 1`] = `
<div class="van-search">
  <div class="van-search__content van-search__content--square">
    <div class="van-cell van-cell--borderless van-field">
      <div class="van-field__left-icon"><i class="van-icon van-icon-setting-o">
          <!----></i></div>
      <div class="van-cell__value van-cell__value--alone van-field__value">
        <div class="van-field__body"><input type="search" class="van-field__control"></div>
      </div>
    </div>
  </div>
</div>
`;

exports[`right-icon prop 1`] = `
<div class="van-search">
  <div class="van-search__content van-search__content--square">
    <div class="van-cell van-cell--borderless van-field">
      <div class="van-field__left-icon"><i class="van-icon van-icon-search">
          <!----></i></div>
      <div class="van-cell__value van-cell__value--alone van-field__value">
        <div class="van-field__body"><input type="search" class="van-field__control">
          <div class="van-field__right-icon"><i class="van-icon van-icon-setting-o">
              <!----></i></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`right-icon slot 1`] = `
<div class="van-search">
  <div class="van-search__content van-search__content--square">
    <div class="van-cell van-cell--borderless van-field">
      <div class="van-field__left-icon"><i class="van-icon van-icon-search">
          <!----></i></div>
      <div class="van-cell__value van-cell__value--alone van-field__value">
        <div class="van-field__body"><input type="search" class="van-field__control">
          <div class="van-field__right-icon">Custom Right Icon</div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
